src/audio/openal/castleopenalsoundbackend.pas

```diff
@@ -274,16 +471,30 @@ procedure TOpenALSoundSourceBackend.SetMaxGain(const Value: Single);
 end;
 
 procedure TOpenALSoundSourceBackend.SetBuffer(const Value: TSoundBufferBackend);
+var
+  I: Integer;
+  Stream: TOpenALStreamBuffersBackend;
+  FullLoaded: TOpenALSoundBufferBackend;
 begin
-  FBuffer := Value as TOpenALSoundBufferBackend;
-  if Value <> nil then
+  FBuffer := Value;
+
+  if Assigned(FBuffer) then
   begin
-    { TSoundBuffer is unsigned, while alSourcei is declared as taking signed integer.
-      But we know we can pass TSoundBuffer to alSourcei, just typecasting it to
-      whatever alSourcei requires. }
-    {$I norqcheckbegin.inc}
-    alSourcei(ALSource, AL_BUFFER, FBuffer.ALBuffer);
-    {$I norqcheckend.inc}
+    if FBuffer is TOpenALSoundBufferBackend then
+    begin
+      FullLoaded := TOpenALSoundBufferBackend(FBuffer);
+      { TSoundBuffer is unsigned, while alSourcei is declared as taking signed integer.
+        But we know we can pass TSoundBuffer to alSourcei, just typecasting it to
+        whatever alSourcei requires. }
+      {$I norqcheckbegin.inc}
+      alSourcei(ALSource, AL_BUFFER, FullLoaded.ALBuffer);
+      {$I norqcheckend.inc}
+    end
+    else if FBuffer is TOpenALStreamBuffersBackend then
+    begin
```

Comment:
Make a newline before `if` here, we usually write code in CGE like this:

```
if xxxx then
begin
  ...
end else 
if yyy then
begin
  ...
end;
```

So the `else` is at the end of the line with `end`, not at the beginning of the line with next `begin`.

Reply:
Sometimes I forget in which line what should be, so usually before PR I check all blocks with the open coding convention page. Because I considered this implementation as an early draft, I skipped this step. Now when I know the direction is OK and you want this, I will polish the code :)